src/biconnectivity/articulation.jl

```diff
@@ -78,3 +78,60 @@ function articulation(g::AbstractGraph)
     end
     return findall(state.articulation_points)
 end
+
+function iterative_articulation(g::SimpleGraph)
+    T = eltype(g)
+    s = Vector{Tuple{T, T, T}}()
+    articulation_points = Set{T}()
+    low = zeros(T, nv(g))
+    pre = zeros(T, nv(g))
+    @inbounds for u in vertices(g)
+        pre[u] != 0 && continue
+        v = u
+        children = 0
+        wi::T = 0
+        w::T = 0
+        cnt::T = 1
+        first_time = true
+        while !isempty(s) || first_time
+            first_time = false
+            if  wi < 1
+                pre[v] = cnt
+                cnt += 1
+                low[v] = pre[v]
+                v_neighbors = outneighbors(g,v)
+                wi = 1
+            else
+                wi,u,v = pop!(s)
+                v_neighbors = outneighbors(g,v)
+                w = v_neighbors[wi]
+                low[v] = min(low[v],low[w])
+                if low[w] >= pre[v] && u != v
+                    push!(articulation_points,v)
+                end
+                wi += 1
+            end
+            while wi <= length(v_neighbors)
+                w = v_neighbors[wi]
+                if pre[w] == 0
+                    if u == v
+                        children += 1
+                    end
+                    push!(s,(wi,u,v))
+                    wi = 0
+                    u = v
+                    v = w
+                    break
+                elseif w != u
+                    low[v] = min(low[v],pre[w])
+                end
+                wi += 1
+            end
+            wi < 1 && continue
+        end
+        if children > 1
+            push!(articulation_points,u)
+        end
+    end
+    return collect(articulation_points)
```

Comment:
do you want to return an iterator instead? It might be nicer.

Reply:
iterator won't work probably as duplicate values occur

Reply:
so I am using a Vector

Reply:
I don't understand why you can't just return `articulation_points` here and leave it to the user to collect it if they need a vector.

Reply:
oh i thought that it was required to return a Vector since the `articulation` function did so.So i should just return the Set as it is ?

Reply:
Ah, yeah, you make a good point. Can you do a graph timing on a larger graph (that obviously  doesn't cause a stack overflow with the original `articulation`)? I'd be in favor of dropping the recursive version if the timing isn't too horrible.

Reply:
Here -
```julia
julia> g = SimpleGraph(50000,200000)
{50000, 200000} undirected simple Int64 graph

julia> @btime articulation(g)
  20.932 ms (8 allocations: 788.95 KiB)
146-element Array{Int64,1}:
   230
   465
   974
  1406
  1714
  1885
  2027
  2030
  2269
  2322
     ⋮
 46133
 47220
 47363
 47740
 47893
 48352
 48661
 48688
 49877

julia> @btime iterative_articulation(g)
  20.607 ms (31 allocations: 3.78 MiB)
146-element Array{Int64,1}:
   230
   465
   974
  1406
  1714
  1885
  2027
  2030
  2269
  2322
     ⋮
 46133
 47220
 47363
 47740
 47893
 48352
 48661
 48688
 49877

```

Reply:
@sbromberger any updates ? Should i transfer this code to articulation and drop the recursive version ?